docs/extensibility.md

```diff
@@ -0,0 +1,122 @@
+import { HelmAddOn, HelmAddOnUserProps } from "../helm-addon";
+
+export class MyProductAddOn extends HelmAddOn {
+
+    readonly options: MyProductAddOnProps; // extends HelmAddOnUserProps
+   
+    ...
+ 
+    @dependable('AwsLoadBalancerControllerAddOn') // depends on AwsLoadBalancerController
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        ...
+    }
+```
+### Helm Add-ons
+
+Helm add-ons are the most common case that generally combines provisioning of a helm chart as well as supporting infrastructure such as wiring of proper IAM policies for the Kubernetes service account, provisioning or configuring other AWS resources (VPC, subnets, node groups). 
+
+In order to provide consistency across all Helm add-ons supplied by the SSP framework all Helm add-ons are implemented as derivatives of the `HelmAddOn` base class and support properties based on `HelmAddOnUserProps`. See the [example extension section](#example-extension) below for more details.
+
+Use cases that are enabled by leveraging the base `HelmAddOn` class:
+
+1. Consistency across all helm based add-on will reduce effort to understand how to apply and configure standard add-on options.
+2. Ability to override helm chart repository can enable leveraging private helm chart repository by the customer and facilitate add-on usage for private EKS clusters.
+3. Extensibility mechanisms available in the SSP framework can allow to intercept helm deployments and leverage GitOps driven add-on configuration.
+
+### Non-helm Add-ons
+
+Add-ons that don't leverage helm but require to install arbitrary Kubernetes manifests will not be able to leverage the benefits provided by the `HelmAddOn` however, they are still relatively easy to implement.
+Deployment of arbitrary kubernetes manifests can leverage the following construct:
+
+```typescript
+import { KubernetesManifest } from "@aws-cdk/aws-eks";
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { loadYaml, readYamlDocument } from "../../utils/yaml-utils";
+
+export class MyNonHelmAddOn implements ClusterAddOn {
+    deploy(clusterInfo: ClusterInfo): void {
+        const cluster = clusterInfo.cluster;
+        // Apply manifest
+        const doc = readYamlDocument(__dirname + '/my-product.yaml');
+        // ... apply any substitutions for dynamic values 
+        const manifest = docArray.split("---").map(e => loadYaml(e));
+        new KubernetesManifest(cluster.stack, "myproduct-manifest", {
+            cluster,
+            manifest,
+            overwrite: true
+        });
+    }
+}
+```
+**Note:** When leveraging this approach consider how customers can apply the add-on for fully private clusters. It may be reasonable to bundle the manifest with the add-on in the npm package. 
+
+## Example Extension
+
+[Example extension](https://github.com/shapirov103/ssp-eks-extension) contains a sample implementation of a FluentBit log forwarder add-on and covers the following aspects of an extension workflow:
```

Comment:
Should we move this example to the patterns repo?

Reply:
Patterns is already overloaded. This extension was something that partners (or customers) are expected to clone and it serves as a ready skeleton. It should be moved under aws-samples though. I will create an issue for that.

Reply:
https://github.com/aws-quickstart/ssp-amazon-eks/issues/270